public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -172,6 +172,22 @@ export function addLabelToQuery(
     )
   );
 
+  // assert that every stream selector doesn't already have this added
+  const everyStreamSelectorAlreadyHasThisAdded = streamSelectorPositions.every((matcherPosition) => {
```

Comment:
Nit, but I find "this" a bit unpleasant to read. Maybe "every stream selector has label" or "all stream selectors processed"